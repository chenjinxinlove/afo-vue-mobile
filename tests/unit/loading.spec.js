import Loading from '../../src/loading/index'
import Vue from 'vue'
import Demo from '../../src/loading/demo'
import {shallowMount, mount} from '@vue/test-utils'

describe('Loading.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Loading)
    expect(Vue.component(Loading.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(Loading, {
      propsData: {
        color: 'white',
        size: '40'
      }
    })
    expect(wrapper.classes()).toEqual(['afo-loading', 'afo-loading-white'])
    expect(wrapper.find('.afo-loading-spinners').attributes().style).toBe('width: 40px; height: 40px;')
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo)).toMatchSnapshot()
  })
})
