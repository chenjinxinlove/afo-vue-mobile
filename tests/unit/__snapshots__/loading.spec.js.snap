// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading.vue demo snapshot 1`] = `
<div>
  <afo-loading-stub size="30" color="white"></afo-loading-stub>
</div>
`;
